### PR TITLE
Rk/uwp

### PR DESCRIPTION
**Pull request by gleblebedev**
@rokups changes + basic gamepad support.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,7 +29,9 @@ cmake_minimum_required (VERSION 3.14)
 cmake_policy(SET CMP0077 NEW)
 
 # Set C++ standard
-set (CMAKE_CXX_STANDARD 17)
+if (NOT DEFINED CMAKE_CXX_STANDARD)
```
Comment (eugeneko) [inline CMakeLists.txt:32]: BTW, why do we need it?

File: Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp [hunk 2]
```diff
@@ -2081,14 +2106,16 @@ bool Graphics::CreateDevice(int width, int height)
         impl_->swapChain_ = nullptr;
     }
 
-    IDXGIDevice* dxgiDevice = nullptr;
-    impl_->device_->QueryInterface(IID_IDXGIDevice, (void**)&dxgiDevice);
+    IDXGIDevice3* dxgiDevice = nullptr;
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp:2109]: This stuff looks dangerous. Does it work well if you get `D3D_FEATURE_LEVEL_10_0`?

File: Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp [hunk 3]
```diff
@@ -2043,13 +2058,23 @@ bool Graphics::CreateDevice(int width, int height)
     // Device needs only to be created once
     if (!impl_->device_)
     {
+        D3D_FEATURE_LEVEL featureLevels[] =
+        {
+            D3D_FEATURE_LEVEL_11_1,
+            D3D_FEATURE_LEVEL_11_0,
+            D3D_FEATURE_LEVEL_10_1,
+            D3D_FEATURE_LEVEL_10_0,
+            D3D_FEATURE_LEVEL_9_3,
```
Comment (eugeneko) [inline Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp:2067]: I don't think D3D_FEATURE_LEVEL_9_3 are supported

File: CMakeLists.txt (hunk 1 above)
Comment (rokups) [reply CMakeLists.txt:32]: So user could use C++20

File: Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp (hunk 2 above)
Comment (rokups) [reply Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp:2109]: Thats how UrhoSharp did it ¯\\\_(ツ)_/¯

File: Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp (hunk 3 above)
Comment (rokups) [reply Source/Urho3D/Graphics/Direct3D11/D3D11Graphics.cpp:2067]: I will test with trimmed feature level list

**Comment by rokups** [comment]
@eugeneko please see last commit in this branch. If its ok, time to merge.

**Pull request merged**